### PR TITLE
[Sentinel] Upgrade Microsoft Sentinel

### PR DESCRIPTION
**Pull request by necusjz**
Resolves: https://github.com/Azure/azure-cli/issues/21665

---
**Powered by CodeGen v2.** Due to current limitation, the following command groups are not generated:
- az sentinel entity;
- az sentinel watchlist item;

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Sentinel

File: src/securityinsight/README.md [hunk 1]
```diff
@@ -1,184 +1,5 @@
...
-    --incident-id "73e01a99-5cd7-4139-a149-9f2736ff2ab5" --resource-group "myRg" --workspace-name "myWorkspace" 
-```
-##### List #####
-```
-az sentinel incident-comment list --incident-id "73e01a99-5cd7-4139-a149-9f2736ff2ab5" --resource-group "myRg" \
-    --workspace-name "myWorkspace" 
-```
\ No newline at end of file
+## How to use ##
+Please add commands usage here.
```
Comment (kairu-ms) [inline src/securityinsight/README.md]: Please complete this file.

File: src/securityinsight/setup.py [hunk 2]
```diff
@@ -24,35 +20,30 @@
...
-except ImportError:
-    pass
-
 with open('README.md', 'r', encoding='utf-8') as f:
     README = f.read()
 with open('HISTORY.rst', 'r', encoding='utf-8') as f:
     HISTORY = f.read()
 
 setup(
-    name='sentinel',
```
Comment (kairu-ms) [inline src/securityinsight/setup.py:46]: Please don't change the extension name

File: src/securityinsight/README.md (hunk 1 above)
Comment (necusjz) [reply src/securityinsight/README.md]: WIP🚧

File: src/securityinsight/azext_securityinsight/azext_metadata.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isExperimental": true,
+    "azext.minCliCoreVersion": "2.37.0"
```
Comment (kairu-ms) [inline src/securityinsight/azext_securityinsight/azext_metadata.json:3]: Please waiting for the release of "2.38.0"
```suggestion
    "azext.minCliCoreVersion": "2.38.0"
```

Comment (necusjz) [reply src/securityinsight/azext_securityinsight/azext_metadata.json:3]: good catch👍

**Review by kairu-ms — APPROVED** [review]
Please release this extension after azure-cli 2.38.0 is released.

**Pull request merged**